src/components/task-plan/homework/add-exercises.cjsx

```diff
@@ -98,6 +98,8 @@ AddExercises = React.createClass
 
     sharedProps =
         exercises: exercises.all
+        watchStore: TaskPlanStore
+        watchEvent: 'change-exercise-'
         onExerciseToggle: @onExerciseToggle
```

Comment:
I'm hopeful that this will allow us to remove my hacky `selectCount={SELECT_COUNT} # re-render flag` stuff on line 121 below.  Can you remove that and see if HW builder still allows you to select exercises?